base_tier_validation_waiting/models/tier_validation.py

```diff
@@ -0,0 +1,48 @@
+# Copyright (c) 2022 brain-tec AG (https://braintec.com)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import fields, models
+
+
+class TierValidation(models.AbstractModel):
+    _inherit = "tier.validation"
+
+    def _notify_review_available(self, tier_reviews):
+        """method to notify when reaching pending"""
+        subscribe = "message_subscribe"
+        post = "message_post"
+        if hasattr(self, post) and hasattr(self, subscribe):
+            for rec in self:
+                users_to_notify = tier_reviews.filtered(
+                    lambda r: r.definition_id.notify_on_pending and r.res_id == rec.id
+                ).mapped("reviewer_ids")
+                # Subscribe reviewers and notify
+                getattr(rec, subscribe)(
+                    partner_ids=users_to_notify.mapped("partner_id").ids
+                )
+                getattr(rec, post)(
```

Comment:
Needs to call with sudo permissions, as the reviewer might not have write permissions on the entity. Similar to https://github.com/OCA/server-ux/blob/14.0/base_tier_validation/models/tier_validation.py#L325